src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/ThreadScheduler.kt

```diff
@@ -119,12 +122,11 @@ open class ThreadScheduler {
      *
      * @return a map from thread ids to thread instances that are currently registered.
      */
-    fun getRegisteredThreads() : ThreadMap<Thread> = mutableThreadMapOf<Thread>().apply {
-        for ((threadId, threadData) in threads) {
+    fun getRegisteredThreads() : Sequence<RegisteredThread> {
```

Comment:
Why do you need to change the API here?
Why `Sequence` instead of `List`?
Why introduce `RegisteredThread`? You can already iterate through pairs `(threadId, thread)` using `ThreadMap` API (thread map is just a typealias to a regular map).

Reply:
it was a memory optimizations: from 560 MB on all tests allocated -> 185 mbs.
reversed as not needed right now